### ISSUE TITLE
[Issue]: ATOM build holds **~16 GB/GPU more resident memory than SGLang without ATOM for the identical model**

### ISSUE THREAD

**Issue by ibeattyo** [ROCm/ATOM#1935]
### Problem Description

This may be related to this: https://github.com/ROCm/ATOM/issues/1929

### Summary
The ATOM build holds **~16 GB/GPU more resident memory than
SGLang without ATOM for the identical model**, almost all of it **outside PyTorch's allocator**
(raw `hipMalloc` kernel/plugin workspace, not model parameters). This is what makes
even plain `dp8_8tp` DP-attention (no EP) OOM on ATOM: at `mem-fraction-static=0.85`
the fixed reservation leaves no headroom for the DP-attention multimem all-gather +
CUDA-graph capture transient.

### Measurement
Instrumented `ModelRunner` to log torch allocator vs driver memory at `Load weight
end` (Qwen3.6-35B-A3B, BF16, TP=8, per GPU, all 8 ranks agree within ~0.4 GB):

| metric | AITER 0.5.17 | ATOM 0.5.15 | Δ (ATOM − AITER) |
|---|---:|---:|---:|
| `torch.cuda.memory_allocated` (live tensors = weights) | 9.3 GB | 11.3 GB | +2.0 GB |
| `torch.cuda.memory_reserved` (torch caching allocator) | 9.4 GB | 11.4 GB | +2.0 GB |
| driver used (`total − mem_get_info().free`) | 19.5 GB | 35.5 GB | +16.0 GB |
| **non-torch** (driver_used − torch_reserved) | 10.1 GB | **24.1 GB** | **+14.0 GB** |

Interpretation: the actual weight **tensors** are only ~2 GB heavier on ATOM. The
dominant term is **~14 GB of extra non-PyTorch device memory** — raw allocations from
kernels/plugins (aiter/CK JIT + ATOM attention/MoE scratch) reserved at init. It is a
**fixed** reservation: measured as a near-constant ~20 GB free-memory delta across
`8tp` / `8tp_8ep` / `dp8_8tp` (sharding-invariant), confirming it is workspace, not
sharded parameters.

### Reproduce the measurement
Add after the `Load weight end` log in
`python/sglang/srt/model_executor/model_runner.py`:
```python
import torch as _t
_resv = _t.cuda.memory_reserved() / 1024**3
_free, _total = _t.cuda.mem_get_info()
logger.info(f"MEM-DIAG torch_alloc={_t.cuda.memory_allocated()/1024**3:.1f} "
            f"torch_reserved={_resv:.1f} driver_used={(_total-_free)/1024**3:.1f} "
            f"non_torch={(_total-_free)/1024**3-_resv:.1f} GB")
```
then launch any TP=8 load. ATOM reports `non_torch≈24 GB`; AITER `≈10 GB`.

### Attribution attempted (needs maintainer-side tooling)
External attribution of the ~14 GB was blocked on the ATOM build and likely needs
someone with the ATOM/kernel source:
- **`LD_PRELOAD` `hipMalloc` interposer** (to log the allocating library per call):
  **segfaults every rank at init** on ATOM. An empty no-op `LD_PRELOAD` runs fine, so
  it is the interposition of the core `hipMalloc` symbol — not `LD_PRELOAD` itself —
  that destabilizes the ATOM HIP runtime (it reroutes libamdhip64's own internal
  calls). This does **not** reproduce on other stacks.
- **`rocprofv3 --memory-allocation-trace`**: wrapping the launcher only traces the
  parent/wrapper processes (a few tiny host allocations) — it does not follow the
  spawned scheduler worker processes where the device allocations happen; a direct
  single-process `Engine(tp_size=1)` load under `rocprofv3` aborts (SIGABRT) during
  init. So the ~14 GB is confirmed to be raw device memory outside torch, but the
  exact call site needs internal instrumentation (roctracer hooks in the ATOM build,
  or a source-level audit of workspace/scratch reservations at model init).

### Impact
- Blocks **DP-attention** on ATOM (both `dp8_8tp` and `dp8_ep8`) — OOM at the
  multimem all-gather / CUDA-graph capture, since SGLang's ~15% non-static headroom
  is consumed by the fixed reservation.
- Lowers max KV capacity on ATOM at high batch even when it doesn't OOM (SGLang
  shrinks the KV pool to fit the heavier "weights").

### Requested change
Identify and reduce the ~14 GB/GPU non-PyTorch workspace reserved at init (e.g. make
it lazy / right-sized / configurable, or reuse a shared pool). Target: bring ATOM's
resident-at-load footprint within a few GB of the upstream/AITER baseline (~10 GB
non-torch), which would let DP-attention run at `mem-fraction-static=0.85`.

### Workaround (until fixed)
Lower `--mem-fraction-static` to ~0.75 (and/or `--chunked-prefill-size 32768`) for
ATOM DP-attention cells to free capture headroom, at the cost of a smaller KV pool.

ibeattyo@smci355-ccs-aus-n14-21:/data/ibeattyo/rocm_testing_framework$ echo "OS:" && cat /etc/os-release | grep -E "^(NAME=|VERSION=)";
  echo "CPU: " && cat /proc/cpuinfo | grep "model name" | sort --unique;
  echo "GPU:" && /opt/rocm/bin/rocminfo | grep -E "^\s*(Name|Marketing Name)";
OS:
NAME="Ubuntu"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
CPU: 
model name      : AMD EPYC 9575F 64-Core Processor
GPU:
  Name:                    AMD EPYC 9575F 64-Core Processor   
  Marketing Name:          AMD EPYC 9575F 64-Core Processor   
  Name:                    AMD EPYC 9575F 64-Core Processor   
  Marketing Name:          AMD EPYC 9575F 64-Core Processor   
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-
  Name:                    gfx950                             
  Marketing Name:          AMD Instinct MI355X                
      Name:                    amdgcn-amd-amdhsa--gfx950:sramecc+:xnack-
      Name:                    amdgcn-amd-amdhsa--gfx9-4-generic:sramecc+:xnack-

### Operating System

22.04.5 LTS (Jammy Jellyfish)

### CPU

AMD EPYC 9575F 64-Core Processor

### GPU

AMD Instinct MI355X

### ROCm Version

gfx950

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

<details>
<summary>rocminfo --support output</summary>

```
Paste output here
```

</details>
